File: Java/MaximumSubarray.java
package com.example;

import java.util.Scanner;

public class MaximumSubarray {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		
		int n = sc.nextInt();
		
		int[] nums = new int[n];
		
		for(int i = 0; i < n; i ++)
		{
			nums[i] = sc.nextInt();
		}
		
		System.out.println(maxSubarray(nums));
	}
	
	public static int maxSubarray(int[] nums)
	{
		int currentSum = nums[0];
		
		int maxSum = nums[0];
		
		for(int i = 1; i < nums.length; i ++)
		{
			currentSum = Math.max(nums[i], currentSum + nums[i]);
			
			maxSum = Math.max(maxSum, currentSum);
		}
		
		return maxSum;
	}

}

File: Python/MaximumSubarray.py
def max_subarray(nums):

    current_sum = nums[0]

    max_sum = nums[0]

    for i in range(len(nums)):

        current_sum = max(nums[i], current_sum + nums[i])

        max_sum = max(max_sum, current_sum)

    return max_sum

nums = list(map(int, input().split()))

print(max_subarray(nums))